### DISCUSSION TITLE
Less MQTT messages please. Is it possible?

### DISCUSSION THREAD

**Discussion by ivanvach** [ESPresense/ESPresense#1130] [General]
Thanks to the developers for the great work and persistency.
Glad to see that espresnse managed to find and report over MQTT my iPhone and my iPad. Is it possible to reduce the number of MQTT messages? I found it in the settings that I can choose only certain devices to be reported to MQTT which is good. However the MQTT traffic is quite busy even for the selected few devices. I do understand that it is necessary for applications that are trying to analyze movement or application that require instant action (like turning lights on). I think the latter is better served by motion sensors that espresnse supports as well. I'm trying to use espresense for the opposite of a motion sensor. I'm trying to have it kick-in when I am in the room but I do not move (say laying on a couch reading). For this very purpose I do not need updates every second or so as it is now. Getting an update once per minute would be more than adequate.  Is it possible to setup espresense that way? Thank You

**Comment by DTTerastar** [comment]
![image](https://github.com/ESPresense/ESPresense/assets/1491145/5d3357db-f640-4c96-a95c-88be8d38a5e6)

That would report only once a minute.

**Comment by heyitsyang** [comment]
On a different, but possibly related area of code, a separate setting for the reporting frequency of GPIO Sensors would be useful.  Many sensors (e.g. temperature, humidity, ambient light) simply don't need to be reported in seconds.  This extra traffic seems an unnecessarily burden for the MQTT server & subscribers.

**Comment by ivanvach** [comment]
@DTTerastar Thanks for the response. Unfortunately it doesn't work. I just set it up to 20000 ( 20 sec.) for testing and the avalanche of MQTT messages doesn't stop. Same device was reported over MQTT in 4 sec. With multiple devices at play I still see quite a bit of messages. Furthermore a message is being sent out to topic ......./telemetry every 15 seconds.

**Comment by ivanvach** [comment]
@heyitsyang Good point. It doesn't make sense at all to constantly report the GPIO status. Best is to report only when change occurs and allow user to select threshold.

**Comment by ivanvach** [comment]
BTW for the presence detection it is a good idea to report only change of distance with user selectable threshold.  For example report to MQTT when apple: xxxx  distance changes more than 1 m.  from previous. This would be a nice option IMHO.

**Reply by DTTerastar** [reply]
It is an option and is in the screenshot.  I set it to 500m to stop it from doing anything.
